src/braket/circuits/gates.py

```diff
@@ -1287,6 +1400,129 @@ def phaseshift(
 Gate.register_gate(PhaseShift)
 
 
+class U(TripleAngledGate):
+    r"""Generalized single-qubit rotation gate.
+
+    Unitary matrix:
+
+        .. math:: \mathtt{U}(\theta, \phi, \lambda) = \begin{bmatrix}
+                \cos{(\theta/2)} & -e^{i \lambda} \sin{(\theta/2)} \\
+                e^{i \phi} \sin{(\theta/2)} & -e^{i (\phi + \lambda)} \cos{(\theta/2)}
+                \end{bmatrix}.
+
+    Args:
+        angle_1 (Union[FreeParameterExpression, float]): theta angle in radians.
+        angle_2 (Union[FreeParameterExpression, float]): phi angle in radians.
+        angle_3 (Union[FreeParameterExpression, float]): lambda angle in radians.
+    """
+
+    def __init__(
+        self,
+        angle_1: Union[FreeParameterExpression, float],
+        angle_2: Union[FreeParameterExpression, float],
+        angle_3: Union[FreeParameterExpression, float],
```

Comment:
Can we add defaults for this gate as we did [with MS](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/696974da3b4b9bb80b09c3a60549193b3935c6f6/src/braket/circuits/gates.py#L3409)? That way we can users can define a [u2 gate](https://docs.quantum.ibm.com/api/qiskit/qiskit.circuit.library.U2Gate) by only providing relevant params.

Reply:
Might be tricky since only the first gate would have a default, so maybe we circle back to this in another pr

Reply:
Maybe we just add documentation that setting angle_1 to pi/2 is equivalent to U2